test/unit/atomic/src/AtomicTest.cpp

```diff
@@ -41,294 +41,663 @@
-            ALPAKA_CHECK(*success, operand == reference);
-        }
-
-        //-----------------------------------------------------------------------------
-        // alpaka::atomic::op::Cas with match
-        {
-            operand = operandOrig;
-            T const compare = operandOrig;
-            T const value = static_cast<T>(4);
-            T const ret =
-                alpaka::atomic::atomicOp<
-                    alpaka::atomic::op::Cas>(
-                        acc,
-                        &operand,
-                        compare,
-                        value);
-            ALPAKA_CHECK(*success, operandOrig == ret);
-            T const reference = value;
-            ALPAKA_CHECK(*success, operand == reference);
-        }
-
-        //-----------------------------------------------------------------------------
-        // alpaka::atomic::op::Cas without match
-        {
-            operand = operandOrig;
-            T const compare = operandOrig + static_cast<T>(1);
-            T const value = static_cast<T>(4);
-            T const ret =
-                alpaka::atomic::atomicOp<
-                    alpaka::atomic::op::Cas>(
-                        acc,
-                        &operand,
-                        compare,
-                        value);
-            ALPAKA_CHECK(*success, operandOrig == ret);
-            T const reference = operandOrig;
-            ALPAKA_CHECK(*success, operand == reference);
-        }
+        testAtomicAdd(acc, success, operandOrig);
+        testAtomicSub(acc, success, operandOrig);
+
+        testAtomicMin(acc, success, operandOrig);
+        testAtomicMax(acc, success, operandOrig);
+
+        testAtomicExch(acc, success, operandOrig);
+
+        testAtomicInc(acc, success, operandOrig);
+        testAtomicDec(acc, success, operandOrig);
+
+        testAtomicAnd(acc, success, operandOrig);
+        testAtomicOr(acc, success, operandOrig);
+        testAtomicXor(acc, success, operandOrig);
+
+        testAtomicCas(acc, success, operandOrig);
     }
 };
 
-BOOST_AUTO_TEST_SUITE(atomic)
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) && BOOST_LANG_CUDA
+//#############################################################################
+template<
+    typename TDim,
+    typename TIdx>
+class AtomicTestKernel<
+    alpaka::acc::AccGpuCudaRt<TDim, TIdx>,
+    int>
+{
+public:
+    //-----------------------------------------------------------------------------
+    ALPAKA_NO_HOST_ACC_WARNING
+    ALPAKA_FN_ACC auto operator()(
+        alpaka::acc::AccGpuCudaRt<TDim, TIdx> const & acc,
+        bool * success,
+        int operandOrig) const
+    -> void
+    {
+        testAtomicAdd(acc, success, operandOrig);
+        testAtomicSub(acc, success, operandOrig);
+
+        testAtomicMin(acc, success, operandOrig);
+        testAtomicMax(acc, success, operandOrig);
+
+        testAtomicExch(acc, success, operandOrig);
+
+        //testAtomicInc(acc, success, operandOrig);
```

Comment:
Why is this commented out? Please add a comment why it is not enabled or remove the code.

Reply:
I have added comments to all those commented out lines, that the operations are not supported.